src/documentation/io/authoring-overview.md

```diff
@@ -10,35 +10,76 @@ permalink: /documentation/io/authoring-overview/
 
 _A guide for users who need to connect to a data store that isn't supported by the [Built-in I/O Transforms]({{site.baseurl }}/documentation/io/built-in/)_
 
-> Note: This guide is still in progress. There is an open issue to finish the guide: [BEAM-1025](https://issues.apache.org/jira/browse/BEAM-1025).
 
 * TOC
 {:toc}
 
 ## Introduction
-TODO
+This is a guide to implementing read and write transforms in the Beam model. These transforms are the way that Beam pipelines import data so it can be processed and then write data to a store.
+
+Reading/writing data in Beam is just a parallel task like everything else. Most of the time you can assemble it from ParDo’s, GBK’s etc.; rarely you will need the more specialized Source and Sink classes for specific features. There are changes coming soon (SplittableDoFn, BEAM-65) that will make Source and Sink unnecessary.
+
+As you work on your I/O Transform, be aware that the Beam community is excited to help those building new I/O Transforms and that there are many examples and helper classes.
 
-## Example I/O Transforms
-TODO
 
 ## Suggested steps for implementers
-TODO
+1. Check out this guide and come up with your design. If you'd like, you can email the Beam dev mailing list with any questions you might have. It's good to check there to see if anyone else is working on the same I/O Transform.
+2. If you are planning to contribute your I/O transform to the Beam community, you'll be going through the normal Beam contribution life cycle - see the [Apache Beam Contribution Guide]({{ site.baseurl }}/contribute/contribution-guide/) for more details.
+3. As you're working on your IO transform, the [PTransform Style Guide]({{ site.baseurl }}/contribute/ptransform-style-guide/) will provide specific information on many implementation details of transform. **Much of the information in the PTransform Style Guide is directly relevant to writing I/O Transforms**.
+
 
 ## Read transforms
-TODO
+Read transforms take data from outside of the Beam pipeline and produce PCollections of data.
+
+For data stores or files types where the data can be read in parallel, you can actually think of the process as a mini-pipeline. This will often consist of two steps:
+1. Splitting the data into parts to be read in parallel
+2. Reading from each of those parts.
```

Comment:
nit: the rest of the doc doesn't have period at the end of the list item, so maybe remove this one